playbooks/common/openshift-master/config.yml

```diff
@@ -69,6 +73,11 @@
       openshift_env:
         openshift_hosted_registry_storage_kind: 'nfs'
     when: openshift_hosted_registry_storage_kind is not defined and groups.oo_nfs_to_config is defined and groups.oo_nfs_to_config | length > 0
+  - openshift_facts:
```

Comment:
you probably want to push this openshift_facts call down into the openshift_metrics role instead of doing it in this playbook.

you will however want to do a set_fact for openshift_hosted_metrics_deploy using the oo_option lookup if openshift_hosted_metrics_deploy is not defined still....